src/js/components/Select/__tests__/__snapshots__/Select-test.js.snap

```diff
@@ -1968,7 +1968,267 @@ exports[`Select disabled 2`] = `
 `;
 
 exports[`Select empty results search 1`] = `
+<body>
```

Comment:
mmmm why we have all this changed with the restrictFocus ?? something seems off to me here

Reply:
@oorestisime Just to clarify, are you asking why this test snapshot changed so much just with the removal of "restrictFocus" from DropButton?

Reply:
@taysea usually when we have snapshot changes it means the fix might not be backward compatible.

Reply:
yeah plus the diff seems unrelated since it is in the select.

Reply:
Okay, let me dig a little deeper to check out what is going on. Thanks for the feedback!

Reply:
After further reading, it seems that in navigation menus, the typical behavior is that "tab" is used to navigate between elements. This differs from the behavior of Select, which allows arrows to navigate between elements. If we want to remain with convention on how keyboard interactions should behave in Menus, then I think this entire PR should be disregarded and the existing grommet behavior should stay. For more, see this article: https://hiddedevries.nl/en/blog/2017-01-29-using-javascript-to-trap-focus-in-an-element

Reply:
@ericsoderberghp we need your input here :)

Reply:
@taysea does it mean that your comment about backward compatibility needs to be refined and it is not really the correct behavior:
>If users were previously using tab to navigate within a menu component, they will now need to use arrow keys. However, arrow keys are the correct behavior.

Reply:
@ShimiSun You're right. Thank you. I've updated now.

Reply:
I am confused now. should we drop this PR?

Reply:
I think we should drop it, but I want @ericsoderberghp to be aware of it

Reply:
I agree. @ShimiSun